battle.go

```diff
@@ -228,3 +238,14 @@ type FightTurn struct {
 func (turn FightTurn) Priority() int {
 	return 0
 }
+
+// An item turn has the a higher priority than any move.
+type ItemTurn struct {
+	item   *Item    // Which item is being consumed
+	target *Pokemon // The target Pokemon for the item
+	move   *Move    // Reference to a move for certain items
+}
+
+func (turn ItemTurn) Priority() int {
+	return PRIORITY_MAX
```

Comment:
Priority should be 1
```suggestion
	return 1
```